### DISCUSSION TITLE
How to solve an equation on first cellcenter only

### DISCUSSION THREAD

**Discussion by Sandip433** [usnistgov/fipy#1046] [Q&A]
![2](https://github.com/usnistgov/fipy/assets/132244694/907ff63c-0082-4950-bd77-928b3fdb1a45)
I want to solve these equations and I want to solve equation (3) only on surface i.e. on the first cellcenter only
I used this but it does not affect the equation and T_ads is solved across the whole domain
T_ads = CellVariable(name="Adsorbate Temperature", mesh=m, value=350.0, hasOld=True)
T_ads.setValue(350.0, where=(x<=dx))

```python

from fipy import *

from fipy.solvers import LinearPCGSolver
from tqdm import tqdm
Lx = 1.0e-7  # m
dx = 1.0e-9  # m
nx = int(Lx / dx)  # number of cells in mesh
T = 2e-12
dt = 4e-15
G = 8.95e17  # W/m^3/K
t_0 = 40e-15  # s
w_0 = 1.24489426e-8  # beam waist in m
J = 28.6  # Fluence J/m^2
gamma1 = 78849  # J/m^3/K^2
gamma2 = 249.14  # J/m^3/K^3
K_0 = 72.0  # W/m/K
E_a = 2.15  # eV
eta_e = 1e12
eta_p = 1e10
neu_ads = 1e13
neu_rc = 1.3e13
k = 8.6173303e-5  # eV/K
h = 4.1357e-15  # eV*s
neu_pfac = 1e13
delay = 0
m=Grid1D(nx=nx, dx=dx)
x = m.cellCenters[0]
time = Variable(0.)
S=(J/(2*t_0*w_0))*numerix.exp(-x/w_0)*(((2/(numerix.exp(time/t_0)+numerix.exp(-time/t_0)))**2)+((2/(numerix.exp((time+delay)/t_0)+numerix.exp(-(time+delay)/t_0)))**2))
T_e = CellVariable(name="Electron Temperature", mesh=m, value=350.0, hasOld=True)
T_p = CellVariable(name="Phonon Temperature", mesh=m, value=350.0, hasOld=True)
T_ads = CellVariable(name="Adsorbate Temperature", mesh=m, value=350.0, hasOld=True)
T_ads.setValue(350.0, where=(x<=dx))
C_p = (2801128 + (-357785 - 2801128) / (1 + (T_p / 62.98191) ** 2.06271) + 278.44328 * T_p)  # J/m^3/K
K_e=K_0*(T_e/T_p)
T_1=1/((neu_ads*(h/k))*numerix.exp((neu_ads*(h/k))/T_ads))
T_2=T_ads*(numerix.exp((neu_ads*(h/k))/T_ads)-1)**2
T_3=eta_e*((neu_rc/neu_ads)/(numerix.exp((neu_rc*(h/k))/T_e)-1)-1/(numerix.exp((neu_ads*(h/k))/T_ads)-1))
T_4=eta_p*((neu_rc/neu_ads)/(numerix.exp((neu_rc*(h/k))/T_p)-1)-1/(numerix.exp((neu_ads*(h/k))/T_ads)-1))
eq_e=TransientTerm(coeff=gamma1+gamma2*T_e, var=T_e)==DiffusionTerm(coeff=K_e, var=T_e)-ImplicitSourceTerm(coeff=G, var=T_e)+ImplicitSourceTerm(coeff=G, var=T_p)+S
eq_p=TransientTerm(coeff=C_p, var=T_p)==ImplicitSourceTerm(coeff=G, var=T_e)-ImplicitSourceTerm(coeff=G, var=T_p)
eq_a=TransientTerm(coeff=1, var=T_ads)==  T_ads * T_1 * T_2 * (T_3 + T_4) 
eq = eq_e & eq_p & eq_a 
t_list = []
results_T_e = []
results_T_p = []
results_T_ads = []
solver = LinearPCGSolver(precon=None, tolerance=1e-10, iterations=2000)
progress_bar = tqdm(total=int(T/dt), desc='Solving PDE')
while time() < T:
    t_list.append(float(time()))
    results_T_e.append(float(T_e[0]))
    results_T_p.append(float(T_p[0]))
    results_T_ads.append(float(T_ads[0]))
    T_e.updateOld()
    T_p.updateOld()
    T_ads.updateOld()
    res = eq.sweep(dt= dt , solver=solver)
    time.setValue(time() + dt)
    progress_bar.update() 
progress_bar.close()
TSVViewer(vars=( T_e,T_p,T_ads)).plot(filename="123 2.txt")
```

**Comment by guyer** [comment]
Is this resolved?

**Comment by Sandip433** [comment]
I multiplied the required equation with (x<dx), but it only stops changing values in other cellcenters . The value is same whether it is solved in first cellcenter or not. I do not know why

**Reply by guyer** [reply]
You multiplied equation (3) (left- and right-hand sides) by (x<dx)? You should only multiply the source in the RHS by that limiting condition:
```python
eq_a=TransientTerm(coeff=1, var=T_ads)==  T_ads * T_1 * T_2 * (T_3 + T_4) * (x < dx)
```

**Comment by Sandip433** [comment]
Sir, I multiplied (x < dx) in the RHS of the equation (3) but the problem is that
[Pd_z=Lx_delay0fs0.8ev1.txt](https://github.com/user-attachments/files/15970003/Pd_z.Lx_delay0fs0.8ev1.txt)
[Pd_z=Lx_delay0fs0.8ev2.txt](https://github.com/user-attachments/files/15970004/Pd_z.Lx_delay0fs0.8ev2.txt)
the first file contains the value of the variables at the end of simulation when the equation (3) is solved in whole domain. The second file contains the value of the variables at the end of simulation when the equation (3) is solved in first cellcenter . The reason I am being cautious that both results give same value for third variable where there is no diffusion in second case. So absolute value of third variable should be higher in second case but it's not.

**Reply by guyer** [reply]
There's no diffusion in the first case either. `eq_a` just solves an ODE at every point in space; there's no coupling with neighbors, via diffusion or anything else.
